test/test_other.py

```diff
@@ -14269,6 +14269,44 @@ def test_windows_batch_file_dp0_expansion_bug(self):
     create_file('build_with_quotes.bat',  f'@"emcc" {test_file("hello_world.c")}')
     self.run_process(['build_with_quotes.bat'])
 
+  @only_windows('Check that directory permissions are properly retrieved on Windows')
+  @requires_node
+  def test_windows_nodefs_execution_permission(self):
+    if self.get_setting('WASMFS'):
+      self.set_setting('FORCE_FILESYSTEM')
+    delete_dir('new-dir')
+    src = r'''
+    #include <assert.h>
+    #include <emscripten.h>
+    #include <sys/stat.h>
+    #include <string.h>
+
+    int main(int argc, char * argv[]) {
+      EM_ASM(
+        FS.mkdir('/working');
+        FS.mount(NODEFS, { root: '{{WORKING_DIR}}' }, '/working');
+        FS.mkdir('/working/new-dir');
+        FS.writeFile('/working/new-dir/test.txt', 'test');
+      );
+
+      int err;
+      struct stat s;
+      memset(&s, 0, sizeof(s));
+      err = stat("/working/new-dir", &s);
+      assert(S_ISDIR(s.st_mode));
+      assert((s.st_mode & S_IXUSR) == S_IXUSR);
+      assert((s.st_mode & S_IXGRP) == S_IXGRP);
+      assert((s.st_mode & S_IXOTH) == S_IXOTH);
+
+      err = stat("/working/new-dir/test.txt", &s);
+      assert((s.st_mode & S_IXUSR) == S_IXUSR);
+      assert((s.st_mode & S_IXGRP) == S_IXGRP);
+      assert((s.st_mode & S_IXOTH) == S_IXOTH);
+    }
+    '''.replace('{{WORKING_DIR}}', Path(self.working_dir).as_posix())
```

Comment:
We need to convert the format of the `working_dir` path to POSIX, otherwise, it doesn't mount NODEFS in the right location.

Reply:
This issue is similar to what's described in Node's `path` reference [Windows vs. POSIX](https://nodejs.org/api/path.html#windows-vs-posix).